### PR TITLE
[BREAKING] dsl package refactor

### PR DESCRIPTION
**Pull request by nitinmohan87**
This refactor makes the service design file import only the top-level `dsl` package instead of top-level `design` and `dsl` and http `design` and `dsl` packages.

### Summary of changes
* Move functions in `http/dsl` package into top-level `dsl` package.
* Move functions in `design` and `http/design` into `expr` package.
* Rename goa v2 `Metadata` to `Meta` (gRPC has a concept of Metadata which will conflict with goa v2 Metadata. Hence this change.)

### What actions should the users take?
* Rename any `Metadata` DSL in your design package to `Meta`
* Import only the top-level dsl package `goa.design/goa/dsl` in your design file. Example:
```
import . "goa.design/goa/dsl"
```
instead of
```
import . "goa.design/goa/http/dsl"
import . "goa.design/goa/http/design"
```
If your design uses any plugins, the plugin DSL functions must be called using the package name. Example:
```
import . "goa.design/goa/dsl"
import "goa.design/plugins/cors"  // Use CORS plugin

var _ = Service("calc", func() {
  Description("The calc service exposes public endpoints that defines CORS policy.")
  cors.Origin("http://127.0.0.1", func() {
    cors.Headers("X-Shared-Secret")
    cors.Methods("GET", "POST")
    cors.Expose("X-Time")
    cors.MaxAge(600)
    cors.Credentials()
  })
})
```

### CODE REVIEW

File: dsl/api.go [hunk 1]
```diff
@@ -63,29 +62,43 @@ func API(name string, fn func()) *design.APIExpr {
...
-	if a, ok := eval.Current().(*design.APIExpr); ok {
-		a.Version = ver
+	if s, ok := eval.Current().(*expr.APIExpr); ok {
+		s.Version = ver
+		return
+	}
+	eval.IncompatibleDSL()
+}
+
+// Contact sets the API contact information.
```
Comment (raphael) [inline dsl/api.go]: Should we add header documentation for these?

File: dsl/api.go [hunk 2]
```diff
@@ -108,97 +121,30 @@ func Version(ver string) {
...
-//
-// TermsOfService takes a single argument which is the TOS text or URL.
-//
-// Example:
-//
-//    var _ = API("github", func() {
-//        TermsOfService("https://help.github.com/articles/github-terms-of-API/"
-//    })
-//
+// TermsOfService describes the API terms of services or links to them.
```
Comment (raphael) [inline dsl/api.go]: Should we keep the header documentation?

File: dsl/response.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+//    - The result type attributes are all mapped to the HTTP response body or gRPC response message.
+//
+// Example:
+//
+//    Method("create", func() {
+//        Payload(CreatePayload)
+//        Result(CreateResult)
+//        Error("an_error")
+//
+//        HTTP(func() {
```
Comment (raphael) [inline dsl/response.go]: Would be cool to add a `GRPC` call as well.

File: examples/cellar/gen/http/storage/client/types.go [hunk 4]
```diff
@@ -285,7 +285,7 @@ func NewShowNotFound(body *ShowNotFoundResponseBody) *storage.NotFound {
 	return v
 }
 
-// Validate runs the validations defined on ShowNotFoundResponseBody
+// Validate runs the validations defined on show_not_foundResponseBody
```
Comment (raphael) [inline examples/cellar/gen/http/storage/client/types.go]: That doesn't look right.... Is that solved in a upcoming PR? or maybe examples need to be regenerated?

File: examples/cellar/gen/http/storage/server/types.go [hunk 5]
```diff
@@ -293,7 +293,7 @@ func NewMultiUpdatePayload(body *MultiUpdateRequestBody, ids []string) *storage.
 	return v
 }
 
-// Validate runs the validations defined on AddRequestBody
+// Validate runs the validations defined on addRequestBody
```
Comment (raphael) [inline examples/cellar/gen/http/storage/server/types.go]: That doesn't look right.... Is that solved in an upcoming PR? or maybe examples need to be regenerated?
(skipping other occurrences of this problem)

File: http/middleware/log.go [hunk 6]
```diff
@@ -15,7 +15,7 @@ type (
 	Logger interface {
 		// Log creates a log entry using a sequence of alternating keys
 		// and values.
-		Log(keyvals ...interface{})
+		Log(keyvals ...interface{}) error
```
Comment (raphael) [inline http/middleware/log.go]: Do you understand why this change? seems we return nil everywhere?

File: dsl/response.go (hunk 3 above)
Comment (nitinmohan87) [reply dsl/response.go]: I thought I would add it along with the gRPC PR.

File: http/middleware/log.go (hunk 6 above)
Comment (nitinmohan87) [reply http/middleware/log.go]: If we use goakit plugin, it will replace the stdlib Logger with gokit's Logger which implements the Logger interface like above. So, we can simply use the gokit logger along with this middleware instead of having a separate adapter for it.

Comment (nitinmohan87) [reply http/middleware/log.go]: Right here - https://github.com/goadesign/plugins/pull/61/files#diff-9aaf94aa7f78e80b8d2568dbc56aef4aR103

File: dsl/api.go [hunk 7]
```diff
@@ -63,29 +62,43 @@ func API(name string, fn func()) *design.APIExpr {
...
-//
-// Version accepts a single string argument.
-//
-// Example:
-//
-//    var _ = API("divider", func() {
-//        Version("1.0")
-//    })
-//
+// Version specifies the API version. One design describes one version.
```
[the remark this answers is not in the document]
Comment (nitinmohan87) [reply dsl/api.go]: Fixed

File: dsl/api.go (hunk 1 above)
Comment (nitinmohan87) [reply dsl/api.go]: Fixed

File: dsl/api.go (hunk 2 above)
Comment (nitinmohan87) [reply dsl/api.go]: Fixed

File: http/middleware/log.go (hunk 6 above)
Comment (raphael) [reply http/middleware/log.go]: Ah yes thank you!
